realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy

```diff
@@ -0,0 +1,181 @@
+
+        // Find all the class names
+        def classNames = getClassNames(inputs)
+
+        // Create and populate the Javassist class pool
+        ClassPool classPool = createClassPool(inputs, referencedInputs)
+
+        logger.info "ClassPool contains Realm classes: ${classPool.getOrNull('io.realm.RealmList') != null}"
+
+        // Find the proxy classes
+        def proxyClasses = classNames
+                .findAll { it.endsWith('RealmProxy') }
+                .collect { classPool.getCtClass(it)}
+        logger.info "Proxy Classes: ${proxyClasses*.name}"
+
+        // Find the model classes
+        def modelClasses = proxyClasses.collect { it.superclass }
+        logger.info "Model Classes: ${modelClasses*.name}"
+
+        // Populate a list of the fields that need to be managed with bytecode manipulation
+        def managedFields = []
+        modelClasses.each {
+            managedFields.addAll(it.declaredFields.findAll {
+                it.getAnnotations()
+                it.getAnnotation(Ignore.class) == null && !Modifier.isStatic(it.getModifiers())
+            })
+        }
+        logger.info "Managed Fields: ${managedFields*.name}"
+
+        // Add accessors to the model classes
+        modelClasses.each {
+            BytecodeModifier.addRealmAccessors(it)
+            def proxyInterface = classPool.get("io.realm.${it.getSimpleName()}RealmProxyInterface")
+            it.addInterface(proxyInterface)
+        }
+
+        // Use accessors instead of direct field access
+        classNames.each {
+            logger.info "  Modifying class ${it}"
+            def ctClass = classPool.getCtClass(it)
+            BytecodeModifier.useRealmAccessors(ctClass, managedFields, modelClasses)
+            ctClass.writeFile(outputProvider.getContentLocation(
+                    'realm', getInputTypes(), getScopes(), Format.DIRECTORY).canonicalPath)
+        }
+
+        def toc = System.currentTimeMillis()
+        logger.info "Realm Transform time: ${toc-tic} milliseconds"
+    }
+
+    /**
+     * Create and populate the Javassist class pool
+     * @param inputs The inputs provided by the Transform API
+     * @param referencedInputs the referencedInputs provided by the Transform API
+     * @return the populated ClassPool instance
+     */
+    private ClassPool createClassPool(Collection<TransformInput> inputs, Collection<TransformInput> referencedInputs) {
+        // Don't use ClassPool.getDefault(). Doing consecutive builds in the same run (e.g. debug+release)
+        // will use a cached object and all the classes will result frozen.
+        ClassPool classPool = new ClassPool(null)
+        classPool.appendSystemPath()
+        classPool.appendClassPath(new LoaderClassPath(getClass().getClassLoader()))
+
+        inputs.each {
+            it.directoryInputs.each {
+                classPool.appendClassPath(it.file.absolutePath)
+            }
+
+            it.jarInputs.stream().each {
```

Comment:
Yes, this is needed since Javassist needs at least `RealmObject` and `RealmList` to be in the class pool.